File: test/cpp/test_12.cpp
#include "cpp_deps/boilerplate.h"
#include "../../src/cpp/code_12.cpp"

void test(Solution& sol, const json& input, const json& output) {
    int num = input["num"].get<int>();
    string expected = output.get<string>();
    string result = sol.intToRoman(num);
    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST("test/test_json/test_12.json");
}

File: test/python/test_12.py
import sys
import os

sys.path.append(os.getcwd())
from .python_deps.boilerplate import Test
from src.python.code_12 import Solution


def input_fetcher(case):
    return case["input"]["num"]


def test():
    test = Test(12, input_fetcher)
    solution = Solution().intToRoman
    test.run_tests(solution)
